application/src/main/java/org/thingsboard/server/service/entitiy/alarm/DefaultTbAlarmService.java

```diff
@@ -221,36 +184,28 @@ public AlarmInfo unassign(Alarm alarm, long unassignTs, User user) throws Things
     }
 
     @Override
-    public void unassignUserAlarms(TenantId tenantId, User user, long unassignTs) {
-        AlarmQueryV2 alarmQuery = AlarmQueryV2.builder().assigneeId(user.getId()).pageLink(new TimePageLink(Integer.MAX_VALUE)).build();
-        try {
-            List<AlarmInfo> alarms = alarmService.findAlarmsV2(tenantId, alarmQuery).get(30, TimeUnit.SECONDS).getData();
-            for (AlarmInfo alarm : alarms) {
-                AlarmApiCallResult result = alarmSubscriptionService.unassignAlarm(tenantId, alarm.getId(), getOrDefault(unassignTs));
-                if (!result.isSuccessful()) {
-                    continue;
-                }
-                if (result.isModified()) {
-                    AlarmComment alarmComment = AlarmComment.builder()
-                            .alarmId(alarm.getId())
-                            .type(AlarmCommentType.SYSTEM)
-                            .comment(JacksonUtil.newObjectNode().put("text", String.format("Alarm was unassigned because user %s - was deleted",
-                                            (user.getFirstName() == null || user.getLastName() == null) ? user.getName() : user.getFirstName() + " " + user.getLastName()))
-                                    .put("userId", user.getId().toString())
-                                    .put("subtype", "ASSIGN"))
-                            .build();
-                    try {
-                        alarmCommentService.saveAlarmComment(alarm, alarmComment, user);
-                    } catch (ThingsboardException e) {
-                        log.error("Failed to save alarm comment", e);
-                    }
-                    notificationEntityService.logEntityAction(alarm.getTenantId(), alarm.getOriginator(), result.getAlarm(),
-                            alarm.getCustomerId(), ActionType.ALARM_UNASSIGNED, user);
-                }
+    public List<AlarmId> unassignDeletedUserAlarms(TenantId tenantId, User user, long unassignTs) {
+        List<AlarmId> totalAlarmIds = new ArrayList<>();
+        PageLink pageLink = new PageLink(100, 0, null, new SortOrder("id", SortOrder.Direction.ASC));
+        while (true) {
+            PageData<AlarmId> pageData = alarmService.findAlarmIdsByAssigneeId(user.getTenantId(), user.getId(), pageLink);
+            List<AlarmId> alarmIds = pageData.getData();
+            if (alarmIds.isEmpty()) {
+                break;
             }
+            processAlarmsUnassignment(tenantId, user, alarmIds, unassignTs);
+            totalAlarmIds.addAll(alarmIds);
+            pageLink = pageLink.nextPageLink();
+        }
+        return totalAlarmIds;
+    }
 
-        } catch (InterruptedException | ExecutionException | TimeoutException e) {
-            throw new RuntimeException(e);
+    @TransactionalEventListener(fallbackExecution = true)
+    public void handleEvent(DeleteEntityEvent<?> event) {
```

Comment:
If we move this handleEvent to the HouseKeeper service, then we eliminate the circular reference and make the code cleaner. Logically, the HouseKeeper service will eventually handle many cases related to DeleteEntityEvent